scripts/badger/registry.py

```diff
@@ -2,20 +2,99 @@
+ALLOWED_BEHAVIORS = [
+  'Compounder',
+  'Helper',
+  'DCA',
+  'Ecosystem',
+  'Ecosystem Helper',
+  'None'
+]
+
+# Add any vaults's addresses that you want to promote or demote
+VAULTS_ARRAY = []
+# Add the new statuses you want to promote or promote each vault to
+NEW_STATUS_ARRAY = []
+
+# promote a batch of vaults given their addresses and new status
+def promote_vaults():
+    if VAULTS_ARRAY and NEW_STATUS_ARRAY:
+        for vault in VAULTS_ARRAY:
+            info = SAFE.badger.registry_v2.productionVaultInfoByVault(vault)
+            new_status = NEW_STATUS_ARRAY[VAULTS_ARRAY.index(vault)]
+            if info[2] < new_status:
+                print(info)
+                SAFE.badger.promote_vault(info[0], info[1], info[3], new_status)
+                assert SAFE.badger.registry_v2.productionVaultInfoByVault(vault)[2] == new_status
+                C.print(f"[green]Promoting {vault} to {new_status}[/green]")
+            else:
+                C.print(f"[red]Promoting {vault} to lower state, current state is {info[2]}[/red]")
+        SAFE.post_safe_tx()
+    else:
+        C.print(f"[red]Make sure to add your vaults and status to the arrays![/red]")
+
+
+# demote a batch of vaults given their addresses and new status
+def demote_vaults():
+    if VAULTS_ARRAY and NEW_STATUS_ARRAY:
+        for vault in VAULTS_ARRAY:
+            info = SAFE.badger.registry_v2.productionVaultInfoByVault(vault)
+            new_status = NEW_STATUS_ARRAY[VAULTS_ARRAY.index(vault)]
+            if info[2] > new_status:
+                SAFE.badger.demote_vault(vault, new_status)
+                assert SAFE.badger.registry_v2.productionVaultInfoByVault(vault)[2] == new_status
+                C.print(f"[green]Demoting {vault} to {new_status}[/green]")
+            else:
+                C.print(f"[red]Demoting to higher state, current state is {info[2]}[/red]")
+        SAFE.post_safe_tx()
+    else:
+        C.print(f"[red]Make sure to add your vaults and status to the arrays![/red]")
+
+
+# change a vault's metadata given a vault address and the new matadata
+# Both the vault address and metadata can be passed as strings from the console.
+# NOTE: If metadata includes spaces, it can be passed using single quotation marks
+# Example of call: brownie run scripts/badger/registry.py update_vault_metadata 
+# 0x37d9D2C6035b744849C15F1BFEE8F268a20fCBd8 'name=auraBAL,protocol=Aura,behavior=None'
+def update_vault_metadata(vault, metadata):
+    # Ensure that the new Behavior is allowed
+    if "behavior=" in metadata:
+        new_behavior = metadata.split("behavior=")[1]
+        if new_behavior not in ALLOWED_BEHAVIORS:
+            C.print(f"[red]Wrong vault behavior! Allowed behaviors: {ALLOWED_BEHAVIORS}[/red]")
+            return
+    else:
+        C.print(f"[red]Wrong metadata format[/red]")
+        return
+
+    info = SAFE.badger.registry_v2.productionVaultInfoByVault(vault)
```

Comment:
in this occasion i think is worthy leveraging the `dict()` capability, mainly to be more explicit on which variable we're interested on, feels more natural to read directly the *'metadata'* than *[3]*. it is probably an opinionated comment, but if you agree would be great to adapt the script to use data all around
```suggestion
    metadata = SAFE.badger.registry_v2.productionVaultInfoByVault(vault).dict()['metadata']
```

Reply:
i like this, makes it more readable. some goes for L33 and L51 then.